parser/src/ast/mod.rs

```diff
@@ -180,3 +180,34 @@ impl MatrixAccess {
         self.col_idx
     }
 }
+
+/// [ColumnAccess] is used to represent a column in the execution trace by segment and column
+/// indexes.
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub struct ColumnAccess {
+    segment_idx: usize,
+    column_idx: usize,
+    row_offset: usize,
+}
+
+impl ColumnAccess {
+    pub fn new(segment_idx: usize, column_idx: usize, row_offset: usize) -> Self {
+        Self {
+            segment_idx,
+            column_idx,
+            row_offset,
+        }
+    }
+
+    pub fn segment_idx(&self) -> usize {
+        self.segment_idx
+    }
+
+    pub fn column_idx(&self) -> usize {
+        self.column_idx
+    }
+
+    pub fn row_offset(&self) -> usize {
+        self.row_offset
+    }
+}
```

Comment:
1. As noted, I think this should be renamed to `IndexedTraceAccess`.
2. I also think it should be moved up in the file so it's right below the `NamedTraceAccess`
3. The comment at the top should also be adjusted for accuracy, since it doesn't mention the row offset